src/java/picard/sam/util/ReadNameParser.java

```diff
@@ -0,0 +1,144 @@
+package picard.sam.util;
+
+import htsjdk.samtools.util.Log;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Small class that provides access to the physical location information about a cluster.
+ * All values should be defaulted to -1 if unavailable.  ReadGroup and Tile should only allow
+ * non-zero positive integers, x and y coordinates may be negative.
+ */
+public class ReadNameParser {
+
+    public static final String DEFAULT_READ_NAME_REGEX = "[a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).*".intern();
+
+    public static final int DEFAULT_OPTICAL_DUPLICATE_DISTANCE = 100;
+
+    private final int[] tmpLocationFields = new int[3]; // for optimization of addLocationInformation
+
+    public String readNameRegex;
+
+    public int opticalDuplicatePixelDistance;
+
+    private Pattern readNamePattern;
+
+    private boolean warnedAboutRegexNotMatching = false;
+
+    private final Log log;
+
+    public ReadNameParser(final Log log) {
+        this.log = log;
+    }
+
+    /**
+     * Method used to extract tile/x/y from the read name and add it to the PhysicalLocationShort so that it
+     * can be used later to determine optical duplication
+     *
+     * @param readName the name of the read/cluster
+     * @param loc the object to add tile/x/y to
+     * @return true if the read name contained the information in parsable form, false otherwise
+     */
+    public boolean addLocationInformation(final String readName, final PhysicalLocation loc) {
+        // Optimized version if using the default read name regex (== used on purpose):
+        if (this.readNameRegex == this.DEFAULT_READ_NAME_REGEX) {
+            final int fields = getLastThreeFields(readName, ':', tmpLocationFields);
```

Comment:
Given that the default regex isn't actually used anywhere, I do not see how it would perturb behaviour. The only way in which it could be used is if someone inputed the same string on the commandline, in that case, with 7 fields, the behavior would be broken, so I think that the regex should represent something that actually works, if not we might as well have it be something like "DEFAULT_THIS_DOES_THE_RIGHT_THING"  :smile: